include/zephyr/devicetree/nvmem.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file
+ * @brief NVMEM Devicetree macro public API header file.
+ */
+
+/*
+ * Copyright (c) 2024, Andriy Gelman
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef INCLUDE_ZEPHYR_DEVICETREE_NVMEM_H_
+#define INCLUDE_ZEPHYR_DEVICETREE_NVMEM_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup devicetree-nvmem Devicetree NVMEM API
+ * @ingroup devicetree
+ * @{
+ */
+
+/**
+ * @brief Get the pointer to the nvmem device from the nvmem-cells property
+ *
+ * Example devicetree fragment:
+ *
+ *	mac_eeprom: mac_eeprom@2 {
+ *		#address-cells = <1>;
+ *		#size-cells = <1>;
+ *		mac_address: mac_address@fa {
+ *			reg = <0xfa 0x06>;
+ *		};
+ *	};
+ *
+ *      eth: ethernet {
+ *		nvmem-cells = <&mac_address>;
+ *		nvmem-cell-names = "mac-address";
+ *	};
+ *
+ * Example usage:
+ *
+ *     DT_NVMEM_DEV_BY_IDX(DT_NODELABEL(eth), 0) // DEVICE_DT_GET(DT_NODELABEL(mac_eeprom))
```

Comment:
Mixing tabs and spaces here

```suggestion
 *	eth: ethernet {
 *		nvmem-cells = <&mac_address>;
 *		nvmem-cell-names = "mac-address";
 *	};
 *
 * Example usage:
 *
 *	DT_NVMEM_DEV_BY_IDX(DT_NODELABEL(eth), 0) // DEVICE_DT_GET(DT_NODELABEL(mac_eeprom))
```